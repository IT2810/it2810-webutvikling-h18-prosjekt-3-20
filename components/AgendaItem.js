import { StyleSheet, Text, View } from 'react-native';
import React from 'react';
import PropTypes from 'prop-types';

const bckColor = '#fff';
const styles = StyleSheet.create({
  item: {
    flex: 1,
    padding: 5,
    backgroundColor: bckColor,
    marginTop: 18,
  },
  time: {
    fontWeight: 'bold',
  },
});

const AgendaItem = ({ item }) => <View style={[styles.item, { height: item.height }]}>
  {item.time && <Text style={styles.time}>{item.time}</Text>}
  <Text>{item.name}</Text>
</View>;

AgendaItem.propTypes = {
  item: PropTypes.shape({
    time: PropTypes.string,
    name: PropTypes.string.isRequired,
  }).isRequired,
};

export default AgendaItem;
