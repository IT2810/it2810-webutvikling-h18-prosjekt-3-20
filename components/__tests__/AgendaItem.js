import 'react-native';
import renderer from 'react-test-renderer';
import React from 'react';
import AgendaItem from '../AgendaItem';

const mockedItems = [
  {
    name: 'Test',
    time: '09:30:00',
  },
  {
    name: 'Test',
  },
];

describe('<AgendaItem/>', () => {
  it('should render without failing', () => {
    const tree = renderer.create(<AgendaItem item={mockedItems[0]}/>).toJSON();

    expect(tree).toMatchSnapshot();
  });
  it('should render without failing', () => {
    const tree = renderer.create(<AgendaItem item={mockedItems[1]}/>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
