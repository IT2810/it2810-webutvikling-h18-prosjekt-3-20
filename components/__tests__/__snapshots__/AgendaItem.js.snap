// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AgendaItem/> should render without failing 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "marginTop": 18,
        "padding": 5,
      },
      Object {
        "height": undefined,
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontWeight": "bold",
      }
    }
  >
    09:30:00
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Test
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={undefined}
  >
    Location
  </Text>
</View>
`;

exports[`<AgendaItem/> should render without failing 2`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "marginTop": 18,
        "padding": 5,
      },
      Object {
        "height": undefined,
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Test
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={undefined}
  >
    Location
  </Text>
</View>
`;
